lib/io/buffered/reader.fz

```diff
@@ -76,9 +76,9 @@ public reader(LM type : mutate, rp Read_Provider, buf_size i32) : effect effect_
   # fill the currently empty buffer with up to max_n bytes
   #
   fill_buffer(max_n i32) unit
-  pre match buffer.get
-        a array => a.is_empty
-        io.end_of_file => false
+  pre safety: match buffer.get
```

Comment:
These here also, rather debug than safety, I think?

Reply:
`safety` should IMHO really protect the system from a crash, so I would go for `debug` here. The same for the following 2x`safety`